src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -6689,9 +6689,6 @@ void QueryAnalyzer::resolveGroupByNode(QueryNode & query_node_typed, IdentifierR
         QueryTreeNodes nullable_group_by_keys;
         for (auto & grouping_sets_keys_list_node : query_node_typed.getGroupBy().getNodes())
         {
-            if (settings.enable_positional_arguments)
-                replaceNodesWithPositionalArguments(grouping_sets_keys_list_node, query_node_typed.getProjection().getNodes(), scope);
-
```

Comment:
It's more likely relates to `expandTuplesInList` that has changed the group by expression list to the argument of function `tuple`. The original group by argument has already been registered as resolved in `replaceNodesWithPositionalArguments`:
```cpp
*node_to_replace = projection_nodes[pos]->clone();
 if (auto it = resolved_expressions.find(projection_nodes[pos]); it != resolved_expressions.end())
 {
       resolved_expressions[*node_to_replace] = it->second;
 }
```
But the new one is not, and it got resolved again and cause the bug. So I think the proper solution is to call `resolveExpressionNodeList` before `expandTuplesInList`.

Reply:
> So I think the proper solution is to call `resolveExpressionNodeList` before `expandTuplesInList`.

The "more" proper solution could also be to recursively register all descendant of `node_to_replace` in `resolved_expressions` after cloning it.

Reply:
seems you are right, but proposed solution then is not quite complete - after tuple expanded its already resolved elements still not in `resolved_expressions` map... I need to think about it...

Reply:
yes, you are right, though recursively registering is not that simple I think - nodes in tuple are already cloned and disconnected from projection nodes... probably correct solution is to expand tuple right at the time we replacing positional arguments...